pkg/api/controller.go

```diff
@@ -3684,14 +3684,34 @@ func (c *Controller) GetMetaRange(w http.ResponseWriter, r *http.Request, reposi
 		return
 	}
 
+	// check if presign
+	if swag.BoolValue(params.Presign) {
+		repo, err := c.Catalog.GetRepository(ctx, repository)
+		if c.handleAPIError(ctx, w, r, err) {
+			return
+		}
```

Comment:
Can we use the response returned from `c.Catalog.GetMetarange` ?
If not please move this before the call to `GetMetarange`

Reply:
The existing (non-presigned) flow doesn't require fetching the repository object. Moving it means adding an additional lookup in other flows. Is there a specific reason to do so?